journal.py

```diff
@@ -102,69 +155,90 @@ def update_entry(request, id):
     request.db.cursor().execute(UPDATE_ENTRY, [title, text, id])
 
 
-def write_entry(request):
-    """ add a new entry to the database"""
-    title = request.params.get('title', None)
-    text = request.params.get('text', None)
-    created = datetime.datetime.utcnow()
-    request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
+# def write_entry(request):
+#     """ add a new entry to the database"""
+#     title = request.params.get('title', None)
+#     text = request.params.get('text', None)
+#     created = datetime.datetime.utcnow()
+#     request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
 
 
-@view_config(route_name='add', request_method='POST')
+@view_config(route_name='add', renderer='json')
 def add_entry(request):
-    try:
-        write_entry(request)
-    except psycopg2.Error:
-        # this will catch any errors generated by the database
-        return HTTPInternalServerError
-    return HTTPFound(request.route_url('home'))
-
+    if request.authenticated_userid:
+        if request.method == 'POST':
+            try:
+                Entry.from_request(request)
+            # This error type needs to be updated since it will now be an SQLA error
+            # type.
+            except psycopg2.Error:
+                # this will catch any errors generated by the database
+                return HTTPInternalServerError
+            return get_newest_entry(request)
+    else:
+        return HTTPForbidden()
 
 # HOME PAGE
 @view_config(route_name='home', renderer='templates/list.jinja2')
 def read_entries(request):
     """return a list of all entries as dicts"""
-    cursor = request.db.cursor()
-    cursor.execute(DB_ENTRIES_LIST)
-    keys = ('id', 'title', 'text', 'created')
-    entries = [dict(zip(keys, row)) for row in cursor.fetchall()]
+    entries = Entry.all()
     for entry in entries:
-        entry['text'] = mdown(entry['text'])
+        entry.text = mdown(entry.text)
```

Comment:
When you update the `text` attribute of your entry, you are editing a database-aware item.  This means that at the successful end of a request, your entry will be persisted to the database with the text now converted to HTML.  That's not what you really want to do, I think.  

Can you come up with a way to get markdown-rendered HTML when you put the value into a template, but leave the `text` attribute of your entries unaltered?